### DISCUSSION TITLE
Could we have a setter on the NormalizedModel or access to its Properties to make adding Lazy properties to DTOs easier?

### DISCUSSION THREAD

**Discussion by jbeales** [spatie/laravel-data#803] [Q&A]
Upgrading to v4, I have run into a problem with my code and the new `NormalizedModel`. My Model & associated DTO have some relations that I have defined on the DTO as Lazy properties, and the DTO's `fromModel()` looks like this:

```
public static function fromModel(ProviderAccount $account)
    {
        $normalizer = new ModelNormalizer();

        return self::from(
            array_merge(
                $normalizer->normalize($account), // returns an array in v3, but a NormalizedModel in v4
                [
                    'managedBy' => Lazy::create(fn () => UserData::collection($account->managedBy)),
                    'listings' => Lazy::create(fn () => ListingData::collection($account->listings)),
                    'ratings' => Lazy::create(fn () => RatingData::collection($account->ratings)),
                ]
            )
        );
    }
```
This lets me add the lazy properties without re-writing all 21 properties on the model, but in v4, with the new `NormalizedModel` it doesn't work, (you can't merge an array and an object!).

If there was a setter on `NormalizedModel` it would be possible to create the `NormalizedModel` then add the lazy properties before returning it. If there was a way to access the properties of the `NormalizedModel` it would be possible to get the properties as an array and merge that with the Lazy properties, then return the merged array as I did in v3.

So my question is, am I missing something? Is there a way to add the Lazy properties to the `NormalizedModel` that I haven't thought of, or is this something that I should be opening a PR to create?

**Comment by rubenvanassche** [comment]
Hi @jbeales,

Normalizers are kinda internal classes so they get changed quite often, I wouldn't depend on them for what they're doing right now in your code. That being said, I've been working on a new data feature which might solve your problem: https://github.com/spatie/laravel-data/pull/831
